File: app/src/main/java/com/snhu/ProjectTwo/utilities/UserInfo.java
package com.snhu.ProjectTwo.utilities;


//@Author Christian Clark
//@Date 8-14-25

//Class to hold the data for a userinfo
public class UserInfo {
    private long _id;
    private long _user;
    private String _date;
    private float _weight;

    public UserInfo(long id, long user, String date, float weight){
        _id = id;
        _user = user;
        _date = date;
        _weight = weight;
    }

    public long getId() {
        return _id;
    }

    public float getWeight() {
        return _weight;
    }

    public long getUser() {
        return _user;
    }

    public String getDate() {
        return _date;
    }
}

File: app/src/main/java/com/snhu/ProjectTwo/utilities/UserInfo.kt
package com.snhu.ProjectTwo.utilities

import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter

//@Author Christian Clark
//@Date 1-9-26

/* Data class that contains the information of one row of the weights table
** Holds a date, weight, and the user it belongs to
*/

data class UserInfo(val id: Long, val user: Long, val date: Long, val weight: Float){

    //gets the date as a formatted as a string
    fun getDateString(): String{
        return Instant.ofEpochMilli(date)
            .atZone(ZoneId.systemDefault())
            .format(DateTimeFormatter.ofPattern("MM/dd/yyyy\nhh:mm"))
    }
}
